gmail-to-salesforce-lead/main.bal

```diff
@@ -0,0 +1,193 @@
+//Re-order the imports alphabetically
+import ballerina/lang.runtime;
+import ballerina/log;
+import ballerina/mime;
+import ballerinax/googleapis.gmail as gmail;
+import ballerinax/openai.chat as openAI;
+import ballerinax/salesforce as sfdc;
+
+type Email record {|
+    string 'from;
+    string subject;
+    string body;
+|};
+
+type Name record {|
+    string firstName__c;
+    string lastName__c;
+|};
+
+type Lead record {|
+    *Name;
+    string email__c;
+    string phoneNumber__c;
+    string company__c;
+    string designation__c;
+|};
+
+configurable string gmailAccessToken = ?;
+configurable string openAIKey = ?;
+configurable string salesforceBaseUrl = ?;
+configurable string salesforceAccessToken = ?;
+
+final string label = "Lead";
+
+public function main() returns error? {
+    while true {
+        Email[] emails = check getEmails(label);
+
+        Lead[] leads = [];
+        from Email email in emails
+        do {
+            Lead|error lead = generateLead(email.'from, email.subject, email.body);
+            if lead is Lead {
+                leads.push(lead);
+            }
+        };
+
+        check addLeadsToSalesforce(leads);
+
+        runtime:sleep(600);
+    }
+}
+
+function getEmails(string label) returns Email[]|error {
+    gmail:Client gmailClient = check new ({auth: {token: gmailAccessToken}});
+
+    string[] labelIdsToMatch = check getLabelIds(gmailClient, [label]);
+    if (labelIdsToMatch.length() == 0) {
+        error e = error("Unable to find any labels to match.");
+        return e;
+    }
+
+    gmail:MailThread[] matchingMailThreads = check getMatchingMailThreads(gmailClient, labelIdsToMatch);
+    removeLabels(gmailClient, matchingMailThreads, labelIdsToMatch);
+
+    gmail:Message[] matchingEmails = getMatchingEmails(gmailClient, matchingMailThreads);
+
+    Email[] emails = [];
+    from gmail:Message message in matchingEmails
+    do {
+        Email|error email = parseEmail(message);
+        if email is Email {
+            emails.push(email);
+        }    
+    };
+    
+    return emails;
+}
+
+function getLabelIds(gmail:Client gmailClient, string[] labelsToMatch) returns string[]|error {
+    gmail:LabelList labelList = check gmailClient->listLabels("me");
+
+    return from gmail:Label label in labelList.labels
+        where labelsToMatch.indexOf(label.name) != ()
+        select label.id;
```

Comment:
```suggestion
    return from gmail:Label {name, id} in labelList.labels
        where labelsToMatch.indexOf(name) != ()
        select id;
```

Reply:
Fixed in [cce2eaf](https://github.com/ballerina-guides/integration-samples/pull/32/commits/cce2eafff5580b9fc4de69c431a8951e6c90b36a)